lxd/container_lxc.go

```diff
@@ -1083,6 +1083,25 @@ func (c *containerLXC) Start() error {
 		return err
 	}
 
+	// If stateful, restore now
+	if stateful && shared.PathExists(c.StatePath()) {
```

Comment:
The problem is that with my current implementation there is no way for the client to know if there is state to be restored on startup, so right now our client always sets stateful to true unless --stateless is passed.

It should be possible to plumb the new IsStateful() I introduced with my other branch to return the right thing though which would allow the client to query for whether there is state available and only set stateful to true in such case, at which point yes, we should fail if there is no state to restore.